rust/core-lib/src/annotations.rs

```diff
@@ -78,6 +83,25 @@ impl AnnotationSlice {
             "n": self.ranges.len()
         })
     }
+
+    pub fn to_annotations(&self, view: &View, text: &Rope) -> Annotations {
+        let last_entry = self.ranges.last().unwrap_or(&[0, 0, 0, 0]);
+        let mut sb = SpansBuilder::new(last_entry[2] + last_entry[3]);
```

Comment:
this has taken me a while to figure out.

As far as I understand it this is equivalent to,

```rust
let span_len = self.ranges.last().map(|anno_range| anno_range.2 + anno_range.3).unwrap_or(0);
// ...
```

which might be clearer?

Reply:
also it isn't totally clear that this logic is right, because it isn't clear that lines & cols are in the same unit space?

Reply:
Oh I forgot to commit and push the most recent change. Lines have to be translated into their offset.